ci(dast-nuclei): fix Could not run nuclei: no valid templates were found

## Relevant files:
- `.github/workflows/.dast-nuclei-cmd-api-server.yaml`
- `.nuclei-config.yaml`


## Logs

[2023-07-12T13-47-05-cacti-ci-dast-nuclei-scan-crash.log](https://github.com/hyperledger/cacti/files/12233106/2023-07-12T13-47-05-cacti-ci-dast-nuclei-scan-crash.log)


```sh
2023-07-12T13:56:05.9053715Z run command "nuclei --config=.nuclei-config.yaml --list=urls.txt --sarif-export ~/nuclei.sarif --output nuclei.log" command "nuclei --config=.nuclei-config.yaml --list=urls.txt --sarif-export ~/nuclei.sarif --output nuclei.log"
2023-07-12T13:56:05.9073349Z [command]/usr/bin/bash -c nuclei --config=.nuclei-config.yaml --list=urls.txt --sarif-export ~/nuclei.sarif --output nuclei.log
2023-07-12T13:56:06.0357314Z 
2023-07-12T13:56:06.0358302Z                      __     _
2023-07-12T13:56:06.0358676Z    ____  __  _______/ /__  (_)
2023-07-12T13:56:06.0358938Z   / __ \/ / / / ___/ / _ \/ /
2023-07-12T13:56:06.0359174Z  / / / / /_/ / /__/ /  __/ /
2023-07-12T13:56:06.0359490Z /_/ /_/\__,_/\___/_/\___/_/   2.6.3
2023-07-12T13:56:06.0359685Z 
2023-07-12T13:56:06.0359833Z 		projectdiscovery.io
2023-07-12T13:56:06.0359998Z 
2023-07-12T13:56:06.0360652Z [[34mWRN[0m] Use with caution. You are responsible for your actions.
2023-07-12T13:56:06.0361564Z [[34mWRN[0m] Developers assume no liability and are not responsible for any misuse or damage.
2023-07-12T13:56:06.0926384Z [[34mINF[0m] nuclei-templates are not installed, installing...
2023-07-12T13:56:06.1147342Z [[31mERR[0m] Could not update templates: Version string empty
2023-07-12T13:56:06.1255844Z [[31mERR[0m] Could not read nuclei-ignore file: open /home/runner/.config/nuclei/.nuclei-ignore: no such file or directory
2023-07-12T13:56:06.1264148Z [[31mERR[0m] Could not find template '/home/runner/nuclei-templates': could not find file: stat /home/runner/nuclei-templates: no such file or directory
2023-07-12T13:56:06.1269731Z [[34mINF[0m] Using Nuclei Engine 2.6.3
2023-07-12T13:56:06.1275013Z [[34mINF[0m] Using Nuclei Templates 
2023-07-12T13:56:06.1293069Z [[1;31mFTL[0m] Could not run nuclei: no valid templates were found
2023-07-12T13:56:06.1453964Z ##[error]The process '/usr/bin/bash' failed with exit code 1
2023-07-12T13:56:06.1539615Z Post job cleanup.
2023-07-12T13:56:06.2674151Z [command]/usr/bin/git version
2023-07-12T13:56:06.2736788Z git version 2.41.0
2023-07-12T13:56:06.2813765Z Temporarily overriding HOME='/home/runner/work/_temp/da419c67-b678-4ee0-aaed-46c3461721cb' before making global git config changes
2023-07-12T13:56:06.2818909Z Adding repository directory to the temporary git global config as a safe directory
2023-07-12T13:56:06.2828803Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/cacti/cacti
2023-07-12T13:56:06.2910010Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-07-12T13:56:06.2951688Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-07-12T13:56:06.3369347Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-07-12T13:56:06.3401129Z http.https://github.com/.extraheader
2023-07-12T13:56:06.3418000Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2023-07-12T13:56:06.3460825Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-07-12T13:56:06.3962444Z Cleaning up orphan processes
2023-07-12T13:56:06.4752032Z Terminate orphan process: pid (11127) (node)
2023-07-12T13:56:06.4869489Z Terminate orphan process: pid (11134) (node)
2023-07-12T13:56:06.4984888Z Terminate orphan process: pid (11145) (sh)
2023-07-12T13:56:06.5097862Z Terminate orphan process: pid (11146) (node)
```

<!-- Edit the body of your new issue then click the ✓ "Create Issue" button in the top right of the editor. The first line will be the issue title. Assignees and Labels follow after a blank line. Leave an empty line before beginning the body of the issue. -->

--- a/.github/workflows/.dast-nuclei-cmd-api-server.yaml
+++ b/.github/workflows/.dast-nuclei-cmd-api-server.yaml
@@ -40,7 +40,7 @@ jobs:
         with:
           go-version: 1.17
 
-      - run: go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@v2.6.3
+      - run: go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@v2.7.3
 
       - run: nuclei --version
 
@@ -83,7 +83,10 @@ jobs:
       - run: echo "$(jq '.authorizationProtocol = "NONE"' .config.json)" > .config.json
 
       - name: Install Fabric connector into the API server
-        run: echo "$(jq '.plugins += [{"packageName":"@hyperledger/cactus-plugin-ledger-connector-fabric", "type":"org.hyperledger.cactus.plugin_import_type.LOCAL", "action":"org.hyperledger.cactus.plugin_import_action.INSTALL",  "options":{"instanceId":"some-unique-fabric-connector-instance-id", "peerBinary":"/fabric-samples/bin/peer", "connectionProfile":"{}", "dockerBinary":"usr/local/bin/docker","cliContainerEnv":{"CORE_PEER_LOCALMSPID":"Org1MSP","CORE_PEER_ADDRESS":"peer0.org1.example.com:7051","CORE_PEER_MSPCONFIGPATH":"/opt/gopath/src/github.com/hyperledger/fabric/peer/crypto/peerOrganizations/org1.example.com/users/Admin@org1.example.com/msp","CORE_PEER_TLS_ROOTCERT_FILE":"/opt/gopath/src/github.com/hyperledger/fabric/peer/crypto/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt","ORDERER_TLS_ROOTCERT_FILE":"/opt/gopath/src/github.com/hyperledger/fabric/peer/crypto/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem"},"discoveryOptions":{"enabled":true,"asLocalhost":true}}}] ' .config.json)" > .config.json
+        #Uncomment line 87 when the fix for the cactus-plugin-ledger-connector-fabric has been published to npm and remove live 89
+        # run: echo "$(jq '.plugins += [{"packageName":"@hyperledger/cactus-plugin-ledger-connector-fabric", "type":"org.hyperledger.cactus.plugin_import_type.LOCAL", "action":"org.hyperledger.cactus.plugin_import_action.INSTALL",  "options":{"instanceId":"some-unique-fabric-connector-instance-id", "peerBinary":"/fabric-samples/bin/peer", "connectionProfile":"{}", "dockerBinary":"usr/local/bin/docker","cliContainerEnv":{"CORE_PEER_LOCALMSPID":"Org1MSP","CORE_PEER_ADDRESS":"peer0.org1.example.com:7051","CORE_PEER_MSPCONFIGPATH":"/opt/gopath/src/github.com/hyperledger/fabric/peer/crypto/peerOrganizations/org1.example.com/users/Admin@org1.example.com/msp","CORE_PEER_TLS_ROOTCERT_FILE":"/opt/gopath/src/github.com/hyperledger/fabric/peer/crypto/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt","ORDERER_TLS_ROOTCERT_FILE":"/opt/gopath/src/github.com/hyperledger/fabric/peer/crypto/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem"},"discoveryOptions":{"enabled":true,"asLocalhost":true}}}] ' .config.json)" > .config.json
+        #Line 89 is a workaround for the cactus-plugin-connector-fabric issue
+        run: echo "$(jq '.plugins += [{"packageName":"@hyperledger/cactus-plugin-ledger-connector-fabric", "type":"org.hyperledger.cactus.plugin_import_type.LOCAL", "action":"org.hyperledger.cactus.plugin_import_action.INSTALL",  "options":{ "packageSrc":"/home/runner/work/cacti/cacti/packages/cactus-plugin-ledger-connector-fabric/", "instanceId":"some-unique-fabric-connector-instance-id", "peerBinary":"/fabric-samples/bin/peer", "connectionProfile":"{}", "dockerBinary":"usr/local/bin/docker","cliContainerEnv":{"CORE_PEER_LOCALMSPID":"Org1MSP","CORE_PEER_ADDRESS":"peer0.org1.example.com:7051","CORE_PEER_MSPCONFIGPATH":"/opt/gopath/src/github.com/hyperledger/fabric/peer/crypto/peerOrganizations/org1.example.com/users/Admin@org1.example.com/msp","CORE_PEER_TLS_ROOTCERT_FILE":"/opt/gopath/src/github.com/hyperledger/fabric/peer/crypto/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt","ORDERER_TLS_ROOTCERT_FILE":"/opt/gopath/src/github.com/hyperledger/fabric/peer/crypto/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem"},"discoveryOptions":{"enabled":true,"asLocalhost":true}}}] ' .config.json)" > .config.json
 
       - name: Install Besu connector into the API server
         run: echo "$(jq '.plugins += [{"packageName":"@hyperledger/cactus-plugin-ledger-connector-besu","type":"org.hyperledger.cactus.plugin_import_type.LOCAL","action":"org.hyperledger.cactus.plugin_import_action.INSTALL","options":{"rpcApiHttpHost":"http://localhost:8545", "rpcApiWsHost":"ws://localhost:8546", "instanceId":"some-unique-besu-connector-instance-id"}}]' .config.json)" > .config.json
@@ -118,7 +121,7 @@ jobs:
         with:
           build: yarn --version
           start: yarn start:api-server
-          command: "nuclei --config=.nuclei-config.yaml --list=urls.txt --sarif-export ~/nuclei.sarif --output nuclei.log"
+          command: "nuclei -update-templates --config=.nuclei-config.yaml --list=urls.txt --sarif-export ~/nuclei.sarif --output nuclei.log"
           wait-on: "https://localhost:4000/api/v1/api-server/healthcheck"
           # wait for 10 minutes for the server to respond
           wait-on-timeout: 120

